node-registrar/pkg/db/farms.go

```diff
@@ -41,9 +43,10 @@ func (db *Database) GetFarm(farmID uint64) (farm Farm, err error) {
 
 func (db *Database) CreateFarm(farm Farm) (uint64, error) {
 	if err := db.gormDB.Create(&farm).Error; err != nil {
-		if errors.Is(err, gorm.ErrDuplicatedKey) {
+		if errors.Is(err, gorm.ErrDuplicatedKey) || strings.Contains(err.Error(), "23505") {
```

Comment:
not sure if checking on the error code would be better

Reply:
tried to match against pq.ErrorCode but it didn't work so I matched strings